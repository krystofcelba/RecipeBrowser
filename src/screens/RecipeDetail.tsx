import React, { Component } from 'react';
import { ScrollView, Image, StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import glamorous from 'glamorous-native';
const { View } = glamorous;

import { Recipe } from 'src/services/api';
import { getRecipe } from 'src/redux/selectors';
import IngredientsList from 'src/components/recipe-detail/IngredientsList';
import StepsList from 'src/components/recipe-detail/StepsList';
import AppTheme from 'src/assets/appTheme';
import { PaddedView, Text, TitleText, namedGlamorous } from 'src/components/common';

const DescriptionContainer = namedGlamorous(PaddedView, 'DescriptionContainer')({
  paddingBottom: 0,
});
const IngredientsContainer = namedGlamorous(PaddedView, 'IngredientsContainer')({
  paddingVertical: AppTheme.defaultPadding / 2,
  marginVertical: AppTheme.defaultPadding / 2,
  paddingRight: AppTheme.defaultPadding * 2.5,
  backgroundColor: AppTheme.lightSecondaryColor,
});
const StepsContainer = namedGlamorous(PaddedView, 'StepsContainer')({
  paddingTop: 0,
  paddingRight: AppTheme.defaultPadding * 2.5,
});

interface Props {
  navigator: any;
  recipeId: number;
  recipe: Recipe;
}

class RecipeDetail extends Component<Props> {
  render() {
    const { recipe: { image, name, description, ingredients, seasonings, steps } } = this.props;
    const imageUri = typeof image === 'string' ? image : image.uri;
    return (
      <View flex={1} testID="recipeDetailScreen">
        <ScrollView>
          <Image style={styles.recipeImage} source={{ uri: imageUri }} resizeMode="cover" />
          <DescriptionContainer>
            <TitleText>{name}</TitleText>
            <Text>{description}</Text>
          </DescriptionContainer>
          <IngredientsContainer>
            <IngredientsList ingredients={ingredients} seasonings={seasonings} />
          </IngredientsContainer>
          <StepsContainer>
            <StepsList steps={steps} />
          </StepsContainer>
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  recipeImage: {
    width: '100%',
    height: 220,
  },
});

const mapStateToProps = (state, { recipeId }) => ({ recipe: getRecipe(state, recipeId) });

export default connect(mapStateToProps)(RecipeDetail);
