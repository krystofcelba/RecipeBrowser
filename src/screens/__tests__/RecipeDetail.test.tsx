import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import RecipeDetail from '../RecipeDetail';

const mockedRecipes = require('src/__mockData__/recipes.json');
const mockedRecipe = mockedRecipes[0];

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {
  recipes: {
    recipes: { [mockedRecipe.id]: mockedRecipe },
  },
};

describe('RecipeDetail component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<RecipeDetail recipeId={mockedRecipe.id} />, {
      context: { store: mockStore(initialState) },
    });
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
