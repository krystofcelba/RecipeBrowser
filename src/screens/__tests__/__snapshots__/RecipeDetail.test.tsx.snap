// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDetail component renders as expected 1`] = `
<glamorous.View
  flex={1}
  testID="recipeDetailScreen"
>
  <ScrollViewMock>
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": "https://nutrition-assets.freeletics.com/recipes/large/mustard_deviled_eggs.jpg",
        }
      }
      style={
        Object {
          "height": 220,
          "width": "100%",
        }
      }
    />
    <DescriptionContainer>
      <TitleText>
        Mustard Deviled Eggs
      </TitleText>
      <Text>
        This buffet classic can also be healthy! Mayonnaise is replaced by cottage cheese. You can blend the filling for more creaminess.
      </Text>
    </DescriptionContainer>
    <IngredientsContainer>
      <IngredientsList
        ingredients={
          Array [
            Object {
              "amount": 4,
              "name": "hard-boiled eggs",
              "unit": "",
            },
            Object {
              "amount": 2,
              "name": "Dijon mustard",
              "unit": "tsp",
            },
            Object {
              "amount": 100,
              "name": "cottage cheese",
              "unit": "g",
            },
            Object {
              "amount": 2,
              "name": "chopped dill",
              "unit": "tbsp",
            },
          ]
        }
        seasonings={
          Array [
            "Paprika powder",
          ]
        }
      />
    </IngredientsContainer>
    <StepsContainer>
      <StepsList
        steps={
          Array [
            "Halve eggs and remove yolk.",
            "Mix yolks with mustard, cottage cheese and half of dill.",
            "Fill egg halves with yolk mixture and garnish with remaining dill.",
          ]
        }
      />
    </StepsContainer>
  </ScrollViewMock>
</glamorous.View>
`;
